lib/webhdfs/client_v1.rb

```diff
@@ -384,6 +430,19 @@ def request(host, port, method, path, op=nil, params={}, payload=nil, header=nil
                     'Response body is empty...'
                   end
 
+        # when delegation token is expired
+        if res.code == '403' and @renew_kerberos_delegation_token_time_hour
+          if message.include?('{"RemoteException":{')
+            detail = JSON.parse(message) rescue nil
+            if detail.dig('RemoteException', 'exception') == 'InvalidToken' and detail.dig('RemoteException', 'message').include?('HDFS_DELEGATION_TOKEN')
```

Comment:
The calls of `.dig` and `.include?` should be guarded by the safe navigation operator (`&.`) because `detail` and `detail.dig(...)` may be `nil`.